### PR TITLE
Antonik_Products_Search_and_Empty_Shopping_Cart_Test

### PR DESCRIPTION
**Pull request by cheesyy183**

### CODE REVIEW

**Comment by cheesyy183** [comment]
pls check also ProductsSearchTest

File: src/main/java/com/opencart/steps/HeaderPageBL.java [hunk 1]
```diff
@@ -131,4 +131,25 @@ public void verifyAddingMacbookToWishList() {
...
+    }
+
+    public ShoppingCartPageBL getShoppingCartPageBL() {
+        return new ShoppingCartPageBL();
+    }
+
+    public void verifyIfShoppingCartDropDownInfoIsEmpty() {
+        String expectedMessage = "0 item(s) - $0.00";
+
+        Assert.assertEquals(headerPage.getShoppingCartDropDownInfo().getText(), expectedMessage);
```
Comment (adomanich) [inline src/main/java/com/opencart/steps/HeaderPageBL.java:146]: Please, add error message

File: src/main/java/com/opencart/steps/HeaderPageBL.java [hunk 2]
```diff
@@ -131,4 +131,25 @@ public void verifyAddingMacbookToWishList() {
...
+    }
+
+    public void verifyShoppingCartDropDownMessage() {
+        String expectedMessage = "Your shopping cart is empty!";
+
+        headerPage.getShoppingCartDropDownButton().click();
+        Assert.assertEquals(headerPage.getShoppingCartDropDownMessage().getText(), expectedMessage);
```
Comment (adomanich) [inline src/main/java/com/opencart/steps/HeaderPageBL.java:153]: Please, add error message

File: src/main/java/com/opencart/steps/ShoppingCartPageBL.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    private ShoppingCartPage shoppingCartPage;
+
+    public ShoppingCartPageBL() {
+        shoppingCartPage = new ShoppingCartPage();
+    }
+
+    public void verifyIfShoppingCartIsEmpty() {
+        String expectedMessage = "Your shopping cart is empty!";
+
+        Assert.assertEquals(shoppingCartPage.getEmptyShoppingCartMessage().getText(), expectedMessage);
```
Comment (adomanich) [inline src/main/java/com/opencart/steps/ShoppingCartPageBL.java:16]: Please, add error message

File: src/main/java/com/opencart/steps/ShoppingCartPageBL.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+public class ShoppingCartPageBL {
+    private ShoppingCartPage shoppingCartPage;
+
+    public ShoppingCartPageBL() {
+        shoppingCartPage = new ShoppingCartPage();
+    }
+
+    public void verifyIfShoppingCartIsEmpty() {
+        String expectedMessage = "Your shopping cart is empty!";
+
```
Comment (adomanich) [inline src/main/java/com/opencart/steps/ShoppingCartPageBL.java:15]: Please, remove extra space

File: src/main/java/com/opencart/steps/HeaderPageBL.java [hunk 5]
```diff
@@ -131,4 +131,25 @@ public void verifyAddingMacbookToWishList() {
...
+
+    public void verifyIfShoppingCartDropDownInfoIsEmpty() {
+        String expectedMessage = "0 item(s) - $0.00";
+
+        Assert.assertEquals(headerPage.getShoppingCartDropDownInfo().getText(), expectedMessage);
+    }
+
+    public void verifyShoppingCartDropDownMessage() {
+        String expectedMessage = "Your shopping cart is empty!";
+
```
Comment (adomanich) [inline src/main/java/com/opencart/steps/HeaderPageBL.java:151]: Please, remove extra space

File: src/main/java/com/opencart/steps/HeaderPageBL.java [hunk 6]
```diff
@@ -131,4 +131,25 @@ public void verifyAddingMacbookToWishList() {
...
+        headerPage.getShoppingCartSpan().click();
+    }
+
+    public ShoppingCartPageBL getShoppingCartPageBL() {
+        return new ShoppingCartPageBL();
+    }
+
+    public void verifyIfShoppingCartDropDownInfoIsEmpty() {
+        String expectedMessage = "0 item(s) - $0.00";
+
```
Comment (adomanich) [inline src/main/java/com/opencart/steps/HeaderPageBL.java:145]: Please, remove extra space

File: src/test/java/com/opencart/tests/ShoppingCartTest/EmptyShoppingCartTest.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+    @Test
+    public void emptyShoppingCartTest() {
+        new Navigation().navigateToURrl(BASE_URL.getValue());
+
+        MainPageBL mainPageBL = new MainPageBL();
+
+        mainPageBL.getHeaderPageBL().clickOnShoppingCartSpan();
+
+        mainPageBL.getHeaderPageBL().getShoppingCartPageBL().verifyIfShoppingCartIsEmpty();
```
Comment (adomanich) [inline src/test/java/com/opencart/tests/ShoppingCartTest/EmptyShoppingCartTest.java:20]: Please, remove extra space

File: src/test/java/com/opencart/tests/ShoppingCartTest/EmptyDropDownInfoShoppingCartTest.java [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+import static com.opencart.enums.URLs.BASE_URL;
+
+public class EmptyDropDownInfoShoppingCartTest extends BaseTest {
+    @Test
+    public void emptyDropDownInfoShoppingCartTest() {
+        new Navigation().navigateToURrl(BASE_URL.getValue());
+
+        MainPageBL mainPageBL = new MainPageBL();
+
+        mainPageBL.getHeaderPageBL().verifyIfShoppingCartDropDownInfoIsEmpty();
```
Comment (adomanich) [inline src/test/java/com/opencart/tests/ShoppingCartTest/EmptyDropDownInfoShoppingCartTest.java:17]: Please, remove extra space

File: src/main/java/com/opencart/pages/HeaderPage.java [hunk 9]
```diff
@@ -48,6 +48,19 @@
     @FindBy(css = "button[onclick=\"wishlist.add('43');\"]")
     private WebElement macbookWishListButton;
 
+    @FindBy(css = "a[title = 'Shopping Cart'] span")
+    private WebElement shoppingCartSpan;
+
+    @FindBy(css = "span#cart-total")
```
Comment (NataliKinash) [inline src/main/java/com/opencart/pages/HeaderPage.java:54]: Make sure you need span here, try to use only id.

File: src/main/java/com/opencart/pages/HeaderPage.java [hunk 10]
```diff
@@ -48,6 +48,19 @@
...
+    private WebElement shoppingCartDropDownInfo;
+
+    @FindBy(css = "p.text-center")
+    private WebElement shoppingCartDropDownMessage;
+
+    @FindBy(css = "button.btn.btn-inverse.btn-block.btn-lg.dropdown-toggle")
+    private WebElement shoppingCartDropDownButton;
```
Comment (NataliKinash) [inline src/main/java/com/opencart/pages/HeaderPage.java:61]: you have the possibility to use id here, try to use the simplest selectors, 
try with smth like this: '#cart .btn'

File: src/main/java/com/opencart/pages/ShoppingCartPage.java [hunk 11]
```diff
@@ -0,0 +1,14 @@
+package com.opencart.pages;
+
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+public class ShoppingCartPage extends BasePage{
+
+    @FindBy(css = "div#content p")
```
Comment (NataliKinash) [inline src/main/java/com/opencart/pages/ShoppingCartPage.java:8]: probably no need to use div here, id will be enough

File: src/main/java/com/opencart/steps/HeaderPageBL.java (hunk 2 above)
Comment (NataliKinash) [inline src/main/java/com/opencart/steps/HeaderPageBL.java:153]: Try to use 3A pattern to arrange your tests

File: src/main/java/com/opencart/pages/HeaderPage.java [hunk 12]
```diff
@@ -51,6 +51,15 @@
     @FindBy(css = "a[title = 'Shopping Cart'] span")
     private WebElement shoppingCartSpan;
 
+    @FindBy(css = "span#cart-total")
+    private WebElement shoppingCartDropDownInfo;
+
+    @FindBy(css = "p.text-center")
+    private WebElement shoppingCartDropDownMessage;
+
+    @FindBy(css = "button.btn.btn-inverse.btn-block.btn-lg.dropdown-toggle")
```
Comment (NataliKinash) [inline src/main/java/com/opencart/pages/HeaderPage.java:60]: Before making the commit try to use shortcuts to format your code, so all the extra spaces will be removed automatically.

File: src/main/java/com/opencart/pages/HeaderPage.java [hunk 13]
```diff
@@ -48,6 +48,19 @@
...
 
+    @FindBy(css = "a[title = 'Shopping Cart'] span")
+    private WebElement shoppingCartSpan;
+
+    @FindBy(css = "span#cart-total")
+    private WebElement shoppingCartDropDownInfo;
+
+    @FindBy(css = "p.text-center")
+    private WebElement shoppingCartDropDownMessage;
+
```
Comment (NataliKinash) [inline src/main/java/com/opencart/pages/HeaderPage.java:59]: Before making the commit try to use shortcuts to format your code, so all the extra spaces will be removed automatically.

File: src/main/java/com/opencart/pages/HeaderPage.java [hunk 14]
```diff
@@ -48,6 +48,19 @@
...
+    @FindBy(css = "button.btn.btn-inverse.btn-block.btn-lg.dropdown-toggle")
```
Comment (NataliKinash) [inline src/main/java/com/opencart/pages/HeaderPage.java:60]: you have the possibility to use id here, try to use the simplest selectors,
try with smth like this: '#cart .btn'

**Pull request closed without merging**